python/tk_multi_workfiles/browser_form.py

```diff
@@ -669,3 +679,37 @@ def _on_step_filter_changed(self, step_list):
         :param step_list: A list of Shotgun Step dictionaries.
         """
         self.step_filter_changed.emit(get_filter_from_filter_list(step_list))
+
+    # Task status filter methods
+    def populate_task_status_list(self):
+        """
+        Populate the task status list from the project schema.
+        """
+        app = sgtk.platform.current_bundle()
+        project_id = app.context.project.get("id")
+        if project_id:
+            logger.debug(
+                f"Task Status Filter: getting statuses from project {project_id}"
+            )
+            task_status_list = app.shotgun._sg.schema_field_read(
+                "Task", "sg_status_list", {"type": "Project", "id": project_id}
+            )["sg_status_list"]["properties"]["valid_values"]["value"]
+
+        task_status_list.insert(0, "ALL")
+        self._my_tasks_form._ui.task_status_combo.addItems(task_status_list)
+
+    def _on_task_status_combo_changed(self, value):
+        """
+        Called when the task status combo box is changed.
+
+        :param value: The selected value on the combo box.
+        """
+        if value == "ALL":
```

Comment:
"ALL" string is used in 2 functions of same class so maybe we can use a new variable in the class scope to be used in this condition and in a previous insert(), because if that string change in future it will need to be changed 2 times